### PR TITLE
Evaluate `import()` argument synchronously

### LINKED ISSUE #84: `import()` argument shouldn't be evaluated in the arrow scope
**Reproduction**

```js
var babelPluginDynamicImportNode = require("babel-plugin-dynamic-import-node");
var babel = require("@babel/core");

var input = `
let i = 0;
import("./file-" + i++ + ".js");
assert(i === 1);

async function fn() {
  await import(await getSource());
}
`

var out = babel.transformSync(input, {
  configFile: false,
  plugins: [babelPluginDynamicImportNode]
});

console.log(out.code);
```

**Output**

```js
let i = 0;
Promise.resolve().then(() => _interopRequireWildcard(require(`${"./file-" + i++ + ".js"}`)));
assert(i === 1);

async function fn() {
  await Promise.resolve().then(() => _interopRequireWildcard(require(`${await getSource()}`)));
}
```

As you can see in the output code, the first assertion fails because `i++` is evaluated asynchronously, and the second one is a syntax error because `await` is inside a non-arrow function.

**Proposed output**

We can pass the import argument in `Promise.resolve`, so that it is in the correct scope:

```js
let i = 0;
Promise.resolve(`${"./file-" + i++ + ".js"}`).then(_ => _interopRequireWildcard(require(_)));
assert(i === 1);

async function fn() {
  await Promise.resolve(`${await getSource()}`).then(_ => _interopRequireWildcard(require(_)));
}
```

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
Fixes #84

### CODE REVIEW

File: test/fixtures/dynamic-argument/expected.js [hunk 1]
```diff
@@ -2,5 +2,5 @@ function _interopRequireWildcard(obj) {}
 
 const MODULE = Object('test-module');
 
-Promise.resolve().then(() => _interopRequireWildcard(require(`${MODULE}`)));
-Promise.resolve().then(() => _interopRequireWildcard(require(`test-${MODULE}`)));
+Promise.resolve(`${MODULE}`).then(_ => _interopRequireWildcard(require(_)));
+Promise.resolve(`test-${MODULE}`).then(_ => _interopRequireWildcard(require(_)));
```
Comment (ljharb) [inline test/fixtures/dynamic-argument/expected.js:6]: all these fixtures should have a trailing newline

**Comment by nicolo-ribaudo** [comment]
I just found a regression with Babel 7, sorry

**Comment by ljharb** [comment]
whoops, k thanks

**Comment by nicolo-ribaudo** [comment]
@ljharb Could you please tag 2.3.1 as `next` (or maybe unpublish it) on npm so that it isn't automatically downloaded? I'm currently unable to add Babel 7 to the tests of this repo because, since this package is a dependency of Babel 7, Babel 7's modules transform is currently completely broken :sweat_smile:

**Comment by StephenEsser** [comment]
> I just found a regression with Babel 7, sorry. I will open a PR soon.
> 
> ```
> Error: No substitution given for "_". If this is not meant to be a
>                 placeholder you may want to consider passing one of the following options to @babel/template:
>                 - { placeholderPattern: false, placeholderWhitelist: new Set(['_'])}
>                 - { placeholderPattern: /^_$/ }Error: Error: No substitution given for "_". If this is not meant to be a
>                 placeholder you may want to consider passing one of the following options to @babel/template:
>                 - { placeholderPattern: false, placeholderWhitelist: new Set(['_'])}
>                 - { placeholderPattern: /^_$/ }
> ```

Stumbled onto this issue after a clean install of a few private repositories. I'm currently locking `babel-plugin-dynamic-import-node` down to 2.3.0 for local development. Our babel 7 projects are currently unable to build on CI. Looks like this is being worked on, I just wanted to reach out and let you guys know it's impact seems to be pretty wide. I'll follow the issue for updates. Thanks for working on a quick fix @nicolo-ribaudo.

```
> babel src --out-dir lib --copy-files

{ Error: Error: No substitution given for "_". If this is not meant to be a
            placeholder you may want to consider passing one of the following options to @babel/template:
            - { placeholderPattern: false, placeholderWhitelist: new Set(['_'])}
            - { placeholderPattern: /^_$/ }
```

**Comment by ljharb** [comment]
@nicolo-ribaudo done, v2.3.0 is now latest.

**Pull request merged**